### PR TITLE
seed: consider if tracking relationship was already established

### PR DESCRIPTION
**Pull request by kim**
When calling `Node::track_project`, it might very well be that the
project/peer is already tracked, which would result in unnecessary
broadcast traffic. Thus, don't emit any events in this case.

Also, change the order of `track` and `replicate` to ensure we
do actually fetch the newly-tracked peer's view. Additionally, we can
skip `replicate` if we already tracked the peer, as subsequent broadcast
messages won't be considered uninteresting.

Ref #734
Signed-off-by: Kim Altintop <kim@eagain.st>

### CODE REVIEW

File: seed/src/lib.rs [hunk 1]
```diff
@@ -440,18 +443,25 @@ impl Node {
...
             .await?
         };
 
         match &result {
-            Ok(()) => {
+            Ok(true) => {
                 tracing::info!("Successfully tracked project {} from peer {}", urn, peer_id);
             },
+            Ok(false) => {
+                tracing::info!("Project {} from peer {} already tracked", urn, peer_id);
```
Comment (kim) [inline seed/src/lib.rs:463]: Not sure, maybe remain silent?

Comment (FintanH) [reply seed/src/lib.rs:463]: Or `debug` if we want to look into this specific call path for some reason

**Review by FintanH — APPROVED** [review]
Nice, that makes sense!

**Review by cloudhead — DISMISSED** [review]
Yes!

**Comment by kim** [comment]
Please pull out your truth tables and verify I got the booleans right

**Pull request merged**